builder.sh

```diff
@@ -8,5 +8,10 @@ cp -r "$ekamSrc" ekam
 chmod -R u+w ekam
 cd ekam
 mkdir -p deps
-ln -s "$capnprotoSrc" deps/capnproto
+cp -r "$capnprotoSrc" deps/capnproto
+chmod u+w deps/capnproto/c++/src/kj -R
+sed -i 's/\/var\/tmp/\/tmp/g' deps/capnproto/c++/src/kj/filesystem-disk-test.c++
+unset NIX_ENFORCE_PURITY
```

Comment:
If this is going to be our solution, I think it could stand a comment explaining what the issues/with links to relevant docs.

Reply:
Well, I can link to that previous issue (and perhaps this one), but I really have no deep understanding of what is going on so far.  I figured this out by noticing that `builder.sh` fails under `nix-shell` but succeeds if the line `. "$stdenv/setup"` is commented out (#2).  So I ran the following script under `nix-shell`

```
set > env-pre
. "$stdenv/setup"
set > env-post
```

... diffed the outputs, and then manually set the environment variables that were different.  I bisected them (started by commenting out half of the lines) and after a few such rounds determined that `NIX_ENFORCE_PURITY` alone is causing the issues.  Interestingly, the linked issue is pretty much the first thing that comes up when searching the internet for that identifier.  I had found that issue previously, but thought it must have been closed for good reason.

Also, while the current workaround is good enough to get ekam to build, I suspect that ekam is unlikely to be able to build anything else unless the user were to `unset NIX_ENFORCE_PURITY` before invoking ekam in that build as well.

Reply:
Well, there are a handful of other expressions in nixpkgs that either `unset NIX_ENFORCE_PURITY` or set it to the empty string.  So maybe my solution is an acceptable one after all.